src/main/xerus/monstercat/MonsterUtilities.kt

```diff
@@ -310,4 +314,65 @@ class MonsterUtilities(checkForUpdate: Boolean): VBox(), JFXMessageDisplay {
 		}
 	}
 	
+	/** Shows a new window with an ImageView of the requested [coverUrl]
+	 * @param coverUrl URL of the cover to download and show
+	 * @param title Title of the window, only useful when decorated
+	 * @param size Height and width in pixel of the window and image
+	 * @param isDecorated True if the window has borders and title bar with close controls
+	 * @param isDraggable True if the window can be dragged by the mouse
+	 * @param closeOnFocusLost Should we close the window if we're out of focus ?
+	 * @param isResizable Allow resizing the window. The image will follow.
+	 */
+	fun viewCover(coverUrl: String, size: Double? = null, title: String = "Cover Art", isDecorated: Boolean = false, isDraggable: Boolean = true, closeOnFocusLost: Boolean = true, isResizable: Boolean = false){
+		val windowSize: Double = size ?: minOf(Screen.getPrimary().visualBounds.width, Screen.getPrimary().visualBounds.height) / 2
+		
+		val pane = StackPane()
+		val largeImage = ImageView()
+		pane.add(Label("Cover loading..."))
+		pane.add(largeImage)
+		
+		val stage = App.stage.createStage(title, pane).apply {
+			height = windowSize
+			width = windowSize
+			if (isResizable) {
+				widthProperty().addListener { _, _, newValue ->
+					largeImage.fitHeight = newValue as Double
+					largeImage.fitWidth = newValue
+				}
+			}
+			this.isResizable = isResizable
+			
+			widthProperty().addListener { _, _, newValue ->
```

Comment:
is there a reason you add this listener twice?

Reply:
Removed in 46b6182eb6f7a6859969136b3bcbfd548962ddbb